lib/performance_platform/repository/session.rb

```diff
@@ -0,0 +1,18 @@
+class PerformancePlatform::Repository::Session < Sequel::Model(:sessions)
+  dataset_module do
+    def stats
+      DB.fetch("
```

Comment:
Oh gosh this is nasty.  Is there a way to record the data in a way which prevents this nastiness?

Reply:
Doing this would require:
A) Something which records that a username was at a specific address for a day
B) Updating this to look at that
C) Purging that data on a regular basis.

Reply:
Yes the current schema makes it difficult to pull these values out.

This replicates exactly what the old backend used to do.  I would deploy this as is first to make sure it still works before improving it.

Could tackle schema improvements when we split out the tables into separate databases maybe.

Reply:
Could this be the opportunity to do this pulling out?

Reply:
Believe that once we've removed the logrequests functionality, the performance platform is the only thing using the sessions table?

Reply:
Given that:

1. it works.
2. it is the exact same SQL that we already use in SQL API.
3. We don't know how to rewrite it to Sequel's DSL quickly.
4. And, anyway, using raw SQL with adapters is [not that uncommon](https://www.python.org/dev/peps/pep-0249/).
5. We've already dragged statistics to another iteration...

Should we attempt to make this refactoring?

Sounds like a lot of system will be touched, all data on development, staging and production machines will need to be migrated to conform with new schema and potentially other uses of current schema which we haven't yet touched will be broken.

I'm okay with the idea to make it better but it'll take time and then we need to clearly point it out to stakeholders.

Reply:
Agreed it will take a while to perform this work.  Just be wary that this query won't scale well in the future.

As a side note, I'm not too worried about raw SQL.  Kinda like it tbh 👅